dbt/include/databricks/macros/materializations/incremental/strategies.sql

```diff
@@ -38,9 +38,56 @@
     {%- endfor -%}
     {%- set dest_cols_csv = dest_columns | join(', ') -%}
     {%- set source_cols_csv = common_columns | join(', ') -%}
-    insert overwrite table {{ target_relation }}
-    {{ partition_cols(label="partition") }}
-    select {{source_cols_csv}} from {{ source_relation }}
+    
+    {%- if (adapter.is_cluster() and adapter.compare_dbr_version(17, 1) >= 0) or (not adapter.is_cluster() and adapter.behavior.use_replace_on_for_insert_overwrite) -%}
+        {%- if not adapter.is_cluster() %}
+            {{ exceptions.warn("insert_overwrite will perform a dynamic insert overwrite. If you depended on the legacy truncation behavior, consider disabling the behavior flag use_replace_on_for_insert_overwrite.") }}
+        {%- endif -%}
+        {{ get_insert_replace_on_sql(source_relation, target_relation, source_cols_csv) }}
+    {%- else -%}
+        {#-- Use legacy DPO INSERT OVERWRITE for older DBR versions and SQL warehouses with behavior flag disabled --#}
+        insert overwrite table {{ target_relation }}
+        {{ partition_cols(label="partition") }}
+        select {{ source_cols_csv }} from {{ source_relation }}
+    {%- endif -%}
+{% endmacro %}
+
+{% macro add_columns_to_list(target_list, columns) %}
+    {#-- Local helper to add columns to the target list, converting string to list if needed --#}
+    {%- if columns -%}
+        {%- if columns is string -%}
+            {%- do target_list.append(columns) -%}
+        {%- else -%}
+            {%- for col in columns -%}
+                {%- do target_list.append(col) -%}
+            {%- endfor -%}
+        {%- endif -%}
+    {%- endif -%}
+{% endmacro %}
+
+{% macro get_insert_replace_on_sql(source_relation, target_relation, source_cols_csv) %}
+    {%- set partition_by = config.get('partition_by') -%}
+    {%- set liquid_clustered_by = config.get('liquid_clustered_by') -%}
```

Comment:
If this is a new config then I would just suggest naming it just `clustered_by`.